### PR TITLE
Adding some Compile DDC tests

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

File: test/compiler_test.dart [hunk 1]
```diff
@@ -48,6 +48,53 @@ void defineTests() {
       });
     });
 
+    test('compileDDC with web', (){
```
Comment (devoncarew) [inline test/compiler_test.dart]: perhaps rename the preceding test `compileDDC simple`?

File: test/compiler_test.dart [hunk 2]
```diff
@@ -48,6 +48,53 @@ void defineTests() {
...
+          .compileDDC(sampleCodeAsync)
+          .then((DDCCompilationResults result) {
+        expect(result.success, true);
+        expect(result.compiledJS, isNotEmpty);
+        expect(result.modulesBaseUrl, isNotEmpty);
+
+        expect(result.compiledJS, contains("define('dartpad_main', ["));
+      });
+    });
+
```
Comment (devoncarew) [inline test/compiler_test.dart]: nit: I would dartfmt this file

Comment (domesticmouse) [reply test/compiler_test.dart]: Done

File: test/compiler_test.dart (hunk 1 above)
Comment (domesticmouse) [reply test/compiler_test.dart]: Done

File: test/compiler_test.dart (hunk 2 above)
Comment (domesticmouse) [reply test/compiler_test.dart]: And done

**Pull request merged**